cmd/terramate/cli/cloud.go

```diff
@@ -885,6 +992,105 @@ func (c *cli) setGitlabCIMetadata(md *cloud.DeploymentMetadata) {
+	md.BitbucketPipelinesCommit = os.Getenv("BITBUCKET_COMMIT")
+	md.BitbucketPipelinesWorkspace = os.Getenv("BITBUCKET_WORKSPACE")
+	md.BitbucketPipelinesRepoSlug = os.Getenv("BITBUCKET_REPO_SLUG")
+	md.BitbucketPipelinesRepoUUID = os.Getenv("BITBUCKET_REPO_UUID")
+	md.BitbucketPipelinesRepoFullName = os.Getenv("BITBUCKET_REPO_FULL_NAME")
+	md.BitbucketPipelinesBranch = os.Getenv("BITBUCKET_BRANCH")
+	md.BitbucketPipelinesTag = os.Getenv("BITBUCKET_TAG")
+	md.BitbucketPipelinesParallelStep = os.Getenv("BITBUCKET_PARALLEL_STEP")
+	md.BitbucketPipelinesParallelStepCount = os.Getenv("BITBUCKET_PARALLEL_STEP_COUNT")
+	md.BitbucketPipelinesPRID = os.Getenv("BITBUCKET_PR_ID")
+	md.BitbucketPipelinesDestinationBranch = os.Getenv("BITBUCKET_PR_DESTINATION_BRANCH")
+	md.BitbucketPipelinesStepUUID = os.Getenv("BITBUCKET_STEP_UUID")
+	md.BitbucketPipelinesDeploymentEnvironment = os.Getenv("BITBUCKET_DEPLOYMENT_ENVIRONMENT")
+	md.BitbucketPipelinesDeploymentEnvironmentUUID = os.Getenv("BITBUCKET_DEPLOYMENT_ENVIRONMENT_UUID")
+	md.BitbucketPipelinesProjectKey = os.Getenv("BITBUCKET_PROJECT_KEY")
+	md.BitbucketPipelinesProjectUUID = os.Getenv("BITBUCKET_PROJECT_UUID")
+	md.BitbucketPipelinesStepTriggererUUID = os.Getenv("BITBUCKET_STEP_TRIGGERER_UUID")
+}
+
+func (c *cli) newBitbucketReviewRequest(pr *bitbucket.PR) *cloud.ReviewRequest {
+	createdAt, err := time.Parse(time.RFC3339, pr.CreatedOn)
+	if err != nil {
+		printer.Stderr.WarnWithDetails("failed to parse PR created_on time", err)
+	}
+	updatedAt, err := time.Parse(time.RFC3339, pr.UpdatedOn)
+	if err != nil {
+		printer.Stderr.WarnWithDetails("failed to parse PR updated_on time", err)
+	}
+	var avatarURL string
+	if pr.Author.Links != nil && pr.Author.Links.Avatar != nil {
+		avatarURL = pr.Author.Links.Avatar.Href
+	}
+
+	var reviewerApprovalCount int
+	var reviewerChangesRequestedCount int
+	var changesRequestedCount int
+	var approvalCount int
+	for _, participant := range pr.Participants {
+		state, ok := participant.State.(string)
+		if !ok {
+			continue
+		}
+
+		switch state {
+		case "changes_requested":
+			changesRequestedCount++
+			if participant.Role == "REVIEWER" {
+				reviewerChangesRequestedCount++
+			}
+		case "approved":
+			approvalCount++
+			if participant.Role == "REVIEWER" {
+				reviewerApprovalCount++
+			}
+		}
+
+	}
+	// TODO(i4k): Bitbucket does not provide a final review decision from the API but we
+	//  can infer it from the reviewers + participants fields.
+	reviewDecision := ""
+	if len(pr.Reviewers) > 0 {
+		reviewDecision = "review_required"
+		if reviewerApprovalCount > 0 {
+			reviewDecision = "approved"
+		}
+		if reviewerChangesRequestedCount > 0 {
+			reviewDecision = "changes_requested"
+		}
+	}
+
+	rr := &cloud.ReviewRequest{
+		Platform:    "bitbucket",
+		Repository:  c.prj.prettyRepo(),
+		URL:         pr.Links.HTML.Href,
+		Number:      pr.ID,
+		Title:       pr.Title,
+		Description: pr.Summary.Raw,
+		CommitSHA:   c.prj.headCommit(),
```

Comment:
I think this is not correct for deployments where it should be the head commit from the branch, not the head of base branch, no?